packages/viem/src/__tests__/index-test.ts

```diff
@@ -34,35 +67,265 @@ describe("createApiKeyAccount", () => {
+        { name: "from", type: "Person" },
+        { name: "to", type: "Person" },
+        { name: "contents", type: "string" },
+      ],
+    } as const;
+
+    const typedData = {
+      account: turnkeyAccount,
+      domain,
+      types,
+      primaryType: "Mail",
+      message: {
+        from: {
+          name: "Cow",
+          wallet: "0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826",
+        },
+        to: {
+          name: "Bob",
+          wallet: "0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB",
+        },
+        contents: "Hello, Bob!",
+      },
+    } as const;
+
+    const signTypedDataSignature = await walletClient.signTypedData(typedData);
+    const verified = await walletClient.verifyTypedData({
+      ...typedData,
+      address: expectedEthAddress,
+      signature: signTypedDataSignature,
+    });
+
+    expect(signTypedDataSignature).toMatch(/^0x/);
+    expect(verified).toBeTruthy();
+  });
+
+  describe("it signs walletconnect v1 payloads, bridged via EIP-1193", () => {
+    // https://docs.walletconnect.com/1.0/json-rpc-api-methods/ethereum
+
+    testCase("Uniswap payload", async () => {
+      const payload: any = {
+        id: 1683062025301507,
+        jsonrpc: "2.0",
+        method: "eth_sendTransaction",
+        params: [
+          {
+            gas: "0x2fe08", // See comment below
+            value: "0xf4240",
+            from: expectedEthAddress, // See comment below
+            to: "0x4648a43b2c14da09fdf82b161150d3f634f40491",
+            data: "0x3593564c000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000a0000000000000000000000000000000000000000000000000000000006451840800000000000000000000000000000000000000000000000000000000000000020b000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000f42400000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000f4240000000000000000000000000000000000000000000000000000000000677493600000000000000000000000000000000000000000000000000000000000000a00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002bb4fbf271143f4fbf7b91a5ded31805e42b2208d6000bb81f9840a85d5af5bf1d1762f925bdaddc4201f984000000000000000000000000000000000000000000",
+          },
+        ],
+      };
+
+      // Looks like here, too, we are subject to similar issues as described below:
+      // See https://github.com/ethers-io/ethers.js/issues/1683
```

Comment:
is this still relevant given we're in viem tests, rather than ethers?

Reply:
i think so, in a certain way given there are some weird behaviors with anvil <> viem. viem expects a signer if the `from` isn't provided; in a real setting, this would be provided via Metamask/a client-side wallet.

if anything, i can comment this out until we're ready to have an example for react-native-viem